src/manifest.c

```diff
@@ -968,11 +986,12 @@ bool changed_includes(struct manifest *old, struct manifest *new)
  *
  * Note: this function should be called after match_manifests().
  */
-int remove_old_deleted_files(struct manifest *m1, struct manifest *m2)
+int remove_deprecated_files(struct manifest *m1, struct manifest *m2, void *compfunc)
 {
 	GList *list1, *list2;
 	struct file *file1, *file2;
 	int count = 0;
+	bool (*compare)(struct file *, struct file *) = compfunc;
```

Comment:
Better check for `compfunc=NULL` before you call it below. Also, would an `int` be better return type for this so you can report/handle errors from a comparison function, like when `file=NULL`?

Reply:
I think it is reasonable to say the thing calling the comparison function should know if the pointer is Null or not. cf strcmp. So in that case bool is fine as a return type.